test/functional/plugin/man_spec.lua

```diff
@@ -137,4 +140,16 @@ describe(':Man', function()
       ]])
     end)
   end)
+
+  describe("quit with q (#18281)", function()
+    before_each(function()
+      helpers.clear()
+    end)
+
+    it("exits with 0 after q", function()
+      -- funcs.system({ nvim_prog, "-u", "NONE", "-i", "NONE", "--headless", "--cmd", cmdline })
```

Comment:
I'm not sure how to handle this.

Reply:
I think this test could be a good model: https://github.com/neovim/neovim/blob/233c41cb8cfe1aa5a8520fe6104adacc53c0ece2/test/functional/autocmd/termxx_spec.lua#L125

create a `VimEnter` handler (autocmd) that prints the buffer contents (which `:Man!` also has activated as a manpage):

```lua
local output = funcs.system({nvim_prog, '--headless', '+autocmd VimLeave * echo "quit works!!"', '+Man!', '+call nvim_input("q")'}, {'manpage contents'})
eq('quit works!!', output)
```

Reply:
Thank you @justinmk!